lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:farmsmart_flutter/chat/ui/viewmodel/ChatResponseViewModel.dart';
+import 'package:farmsmart_flutter/model/bloc/Transformer.dart';
+import 'package:intl/intl.dart';
+
+class _Strings {
+  static const dateFormat = "MMMM";
+}
+
+class ChatResponseToPlotInfoTransformer extends ObjectTransformer<Map<String,ChatResponseViewModel>, Map<String,String>> {
+
+  final dateFormat = DateFormat(_Strings.dateFormat);
+
+  @override
+  Map<String,String> transform({Map<String,ChatResponseViewModel> from}) {
+    return from.map((key,value){
+      final date = castOrNull<DateTime>(value.value);
+      if (date != null){
+        return MapEntry(key, dateFormat.format(date));
+      }
+      return MapEntry(key, value.id);
```

Comment:
Another way to do that:
```
val mapValue = (date != null) ? dateFormat.format(date) : value.id;
return MapEntry(key, mapValue);
```

Take the one you like most ;-)

Reply:
this will expand with more types I expect, so I leave longhand for now